sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDao.java

```diff
@@ -474,8 +474,19 @@ private void onCaseChanged(Case existingCase, Case changedCase) {
 				}
 			}
 
+			boolean responsibleRegionChanged = !DataHelper.isSame(changedCase.getResponsibleRegion(), existingCase.getResponsibleRegion());
+			boolean regionChanged = !DataHelper.isSame(changedCase.getRegion(), existingCase.getRegion());
+
+			boolean responsibleDistrictChanged = !DataHelper.isSame(changedCase.getResponsibleDistrict(), existingCase.getResponsibleDistrict());
+			boolean districtChanged = !DataHelper.isSame(changedCase.getDistrict(), existingCase.getDistrict());
+
+			boolean responsibleCommunityChanged = !DataHelper.isSame(changedCase.getResponsibleCommunity(), existingCase.getResponsibleCommunity());
+			boolean communityChanged = !DataHelper.isSame(changedCase.getCommunity(), existingCase.getCommunity());
+
+			boolean facilityChanged = !DataHelper.isSame(changedCase.getHealthFacility(), existingCase.getHealthFacility());
+
 			// If the district has changed, assign a new surveillance officer and re-assign tasks
-			if (!DataHelper.isSame(changedCase.getResponsibleDistrict(), existingCase.getResponsibleDistrict())
+			if (responsibleDistrictChanged
 				|| (changedCase.getResponsibleDistrict() == null && !DataHelper.isSame(changedCase.getDistrict(), existingCase.getDistrict()))) {
```

Comment:
This code does not apply to the current logic anymore; the responsible district of a case can never be null. Also, please make sure to ONLY re-assign the surveillance officer of the case here if the current responsible officer's district is NEITHER the responsible district NOR the place of stay district anymore